### PR TITLE
[Kernel scoping 3/5] Add selectors to tracepoint deployment

### LINKED ISSUE #1582: `px/tcp_drops` pxl script broken due to missing tcp_drop kprobe
Running the `px/tcp_drops` pxl script results in the following errors:

```
$ kubectl -n pl logs vizier-pem-qrn9l

[ .. ]

stdin:26-28: WARNING: tcp_drop is not traceable (either non-existing, inlined, or marked as "notrace"); attaching to it will likely fail
I20230627 17:32:19.979256 1894345 stirling.cc:549] DynamicTraceConnector [DT_e2a6ba86-df73-4975-a291-fd8728296097] created in 5570.15 ms.
I20230627 17:32:19.979339 1894345 source_connector.cc:35] Initializing source connector: DT_e2a6ba86-df73-4975-a291-fd8728296097
cannot attach kprobe, probe entry may not exist
ERROR: Error attaching probe: 'kprobe:tcp_drop'
I20230627 17:32:20.119061 1894345 stirling.cc:566] Internal : Failed to run BPF code.
I20230627 17:32:22.009603 3168351 tracepoint_manager.cc:180] Tracepoint[e2a6ba86-df73-4975-a291-fd8728296097]::tcp_drop_tracer has transitioned PENDING_STATE -> FAILED_STATE
```

It appears that the `tcp_drop` function was removed in v5.19 kernel and that removal may have been backported to other v5.x releases. For these newer kernels there is a tracepoint available that should serve a similar purpose (`tracepoint:skb:kfree_skb`). BCC's `tcpdrop` tool uses this as a [fallback](https://github.com/iovisor/bcc/blob/6ef0f0dd2c8d315e2b6dda26e63d014a49ca9ec1/tools/tcpdrop.py#L228-L232) if the kprobe doesn't exist, so we should implement a similar fallback mechanism.

### PR DESCRIPTION
**Pull request by benkilimnik**
Summary: Adds selectors to `TracepointDeployment` and modifies query broker to deploy tracepoints only to agents with matching selectors. Currently intended to fix #1582, but can capture other restrictions (OS version etc.) in the future.

Related issues: #1582 #1659

Type of change: /kind feature

Test Plan: Tested all existing targets and tested selection based on kernel version in `src/vizier/services/metadata/controllers/tracepoint/tracepoint_test.go`

### CODE REVIEW

**Review by ddelnano — COMMENTED** [review]
A few minor comments, but otherwise looks good.

File: src/vizier/services/metadata/controllers/tracepoint/tracepoint.go [hunk 1]
```diff
@@ -295,26 +297,158 @@ func (m *Manager) UpdateAgentTracepointStatus(tracepointID *uuidpb.UUID, agentID
...
+		}
+		msg, err := tracepointReq.Marshal()
+		if err != nil {
+			return err
+		}
+
+		agentIDs := make([]uuid.UUID, len(validAgentsForProgram))
+		for i, agt := range validAgentsForProgram {
+			agentIDs[i] = utils.UUIDFromProtoOrNil(agt.Info.AgentID)
+		}
```
[the remark this answers is not in the document]
Comment (benkilimnik) [reply src/vizier/services/metadata/controllers/tracepoint/tracepoint.go]: Added!

File: src/vizier/services/metadata/controllers/tracepoint/tracepoint.go [hunk 2]
```diff
@@ -295,26 +297,166 @@ func (m *Manager) UpdateAgentTracepointStatus(tracepointID *uuidpb.UUID, agentID
...
+		if err != nil {
+			return err
+		}
+
+		agentIDs := validAgentsForProgram.AgentIDs
+		for i, agt := range validAgentsForProgram.Agents {
+			agentIDs[i] = utils.UUIDFromProtoOrNil(agt.Info.AgentID)
+		}
+
+		err = m.agtMgr.MessageAgents(agentIDs, msg)
```
Comment (ddelnano) [inline src/vizier/services/metadata/controllers/tracepoint/tracepoint.go:453]: Isn't this for loop redundant now? I was hoping we would be able to remove it entirely and write the following:

```go
err = m.agtMgr.MessageAgents(validAgentsForProgram.AgentIDs, msg)
```

Comment (benkilimnik) [reply src/vizier/services/metadata/controllers/tracepoint/tracepoint.go]: Yes sorry, I forgot to remove that loop. Should be fixed now.

**Review by ddelnano — APPROVED** [review]

**Comment by ddelnano** [comment]
@pixie-io/maintainers this is ready for review now.

**Review by JamesMBartlett — APPROVED** [review]

**Pull request merged**